include/moveit_grasps/grasp_filter.h

```diff
@@ -155,13 +160,77 @@ class GraspFilter
    * \param grasp_candidates - all possible grasps that this will test. this vector is returned modified
    * \param arm_jmg - the arm to solve the IK problem on
    * \param filter_pregrasp -whether to also check ik feasibility for the pregrasp position
-   * \return some grasps remaining
+   * \param target_object_id - The name of the target grasp object to be used in the planning scene.
+   * \return if some grasps are still valid
    */
   virtual bool filterGrasps(std::vector<GraspCandidatePtr>& grasp_candidates,
                             const planning_scene_monitor::PlanningSceneMonitorPtr& planning_scene_monitor,
                             const robot_model::JointModelGroup* arm_jmg, const moveit::core::RobotStatePtr& seed_state,
-                            bool filter_pregrasp = false);
+                            bool filter_pregrasp = false, const std::string& target_object_id = "");
 
+  virtual bool filterGrasps(std::vector<GraspCandidatePtr>& grasp_candidates,
+                            const planning_scene::PlanningScenePtr& planning_scene,
+                            const robot_model::JointModelGroup* arm_jmg, const moveit::core::RobotStatePtr& seed_state,
+                            bool filter_pregrasp = false, const std::string& target_object_id = "");
+  /**
+   * \brief Return grasps that are kinematically feasible
+   * \param grasp_candidates - all possible grasps that this will test. this vector is returned modified
+   * \param arm_jmg - the arm to solve the IK problem on
+   * \param filter_pregrasp -whether to also check ik feasibility for the pregrasp position
+   * \param visualize - visualize IK filtering
+   * \param target_object_id - The name of the target grasp object to be used in the planning scene.
+   * \return number of grasps remaining
+   */
+  virtual std::size_t filterGraspsHelper(std::vector<GraspCandidatePtr>& grasp_candidates,
+                                         const planning_scene_monitor::PlanningSceneMonitorPtr& planning_scene_monitor,
+                                         const robot_model::JointModelGroup* arm_jmg,
+                                         const moveit::core::RobotStatePtr& seed_state, bool filter_pregrasp,
+                                         bool visualize, const std::string& grasp_target_object_id = "");
+
+  virtual std::size_t filterGraspsHelper(std::vector<GraspCandidatePtr>& grasp_candidates,
+                                         const planning_scene::PlanningScenePtr& planning_scene,
+                                         const robot_model::JointModelGroup* arm_jmg,
+                                         const moveit::core::RobotStatePtr& seed_state, bool filter_pregrasp,
+                                         bool visualize, const std::string& grasp_target_object_id = "");
+
+  /**
+   * \brief Print grasp filtering statistics
+   */
+  virtual void printFilterStatistics(std::vector<GraspCandidatePtr>& grasp_candidates);
```

Comment:
const ref, and maybe
`const std::vector<GraspCandidateConstPtr>&`  also, this should probably be a const function